### PR TITLE
clearer packaging error messages

### LINKED ISSUE #2951: `npm run package` isn't playing nice without typescript
### Describe the bug

Running `npm run package` on a code stack meant to spit out _javascript modules_ (not typescript) gives us the error:
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'svelte2tsx' imported from /Users/x/Projects/crow/node_modules/@sveltejs/kit/dist/chunks/index6.js
```

Installning `svelte2tsx` makes it work, but it bundles `.d.ts`-files.

### Reproduction

1. Download `npm init svelte@next test`
  (Skeleton Project > **No Typescript** > **Use ESLint** > **Use Prettier**)
2. Install `cd test; npm install`
3. Run `npm run package`

Yields the error: `Cannot find package 'svelte2tsx'`

### Logs

_No response_

### System Info

```shell
System:
  OS: macOS 11.1
  CPU: (8) x64 Apple M1
  Memory: 648.46 MB / 16.00 GB
  Shell: 5.8 - /bin/zsh
Binaries:
  Node: 14.16.0 - ~/.nvm/versions/node/v14.16.0/bin/node
  npm: 7.20.3 - ~/.nvm/versions/node/v14.16.0/bin/npm
Browsers:
  Chrome: 96.0.4664.55
  Firefox: 94.0.2
  Safari: 14.0.2
npmPackages:
  @sveltejs/adapter-vercel: ^1.0.0-next.23 => 1.0.0-next.23
  @sveltejs/kit: ^1.0.0-next.120 => 1.0.0-next.120
  svelte: ^3.42.6 => 3.44.0
```

### Severity

annoyance

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ignatiusmb**
Closes #2951

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/packaging/index.js [hunk 1]
```diff
@@ -326,8 +331,7 @@ async function try_load_svelte2tsx() {
 			return await import('svelte2tsx');
 		} catch (e) {
 			throw new Error(
-				'You need to install svelte2tsx and typescript if you want to generate type definitions\n' +
-					e
+				'You need svelte2tsx and typescript if you want to generate type definitions. Install it through your package manager, or disable generation which are highly discouraged. See https://kit.svelte.dev/docs#packaging'
```
Comment (bluwy) [inline packages/kit/src/packaging/index.js:334]:
```suggestion
				'You need svelte2tsx and typescript if you want to generate type definitions. Install it through your package manager, or disable generation which is highly discouraged. See https://kit.svelte.dev/docs#packaging'
```

Should this be an "is"?

**Review by bluwy — APPROVED** [review]
LGTM. With a small question.

File: packages/kit/src/packaging/index.js (hunk 1 above)
Comment (ignatiusmb) [reply packages/kit/src/packaging/index.js:334]: Yeah, I think I mixed up the wording somewhere in between 😄

File: packages/kit/src/packaging/index.js [hunk 2]
```diff
@@ -12,18 +12,23 @@ const essential_files = ['README', 'LICENSE', 'CHANGELOG', '.gitignore', '.npmig
  * @param {string} cwd
  */
 export async function make_package(config, cwd = process.cwd()) {
-	const package_dir = config.kit.package.dir;
-	const abs_package_dir = path.isAbsolute(package_dir) ? package_dir : path.join(cwd, package_dir);
-	rimraf(abs_package_dir);
-	mkdirp(abs_package_dir); // TODO https://github.com/sveltejs/kit/issues/2333
+	if (!fs.existsSync(config.kit.files.lib)) {
+		throw new Error(`${config.kit.files.lib} does not exists`);
```
Comment (dummdidumm) [inline packages/kit/src/packaging/index.js:16]:
```suggestion
		throw new Error(`${config.kit.files.lib} does not exist`);
```

File: packages/kit/src/packaging/test/errors/no-lib-folder/package.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+	"name": "error-no-lib-folder",
+	"version": "1.0.0",
+	"description": "validate no lib folder error output"
```
Comment (dummdidumm) [inline packages/kit/src/packaging/test/errors/no-lib-folder/package.json:4]: I think we need to add `private:true` if I read the Discord chat about publishing correctly

File: packages/kit/src/packaging/test/index.js [hunk 4]
```diff
@@ -75,25 +75,43 @@ function format(file, content) {
...
+			assert.instance(error, Error);
+			switch (dir) {
+				case 'duplicate-export':
+					assert.match(
+						error.message,
+						'Duplicate "./utils" export. Please remove or rename either $lib/utils/index.js or $lib/utils.ts'
+					);
+					break;
+				case 'no-lib-folder':
+					assert.match(error.message, `${join(cwd, 'src', 'lib')} does not exists`);
```
Comment (dummdidumm) [inline packages/kit/src/packaging/test/index.js:97]:
```suggestion
					assert.match(error.message, `${join(cwd, 'src', 'lib')} does not exist`);
```

File: packages/kit/src/packaging/test/errors/duplicate-export/package.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-	"name": "errors",
+	"name": "error-duplicate-export",
 	"version": "1.0.0",
-	"description": "validate error output"
+	"description": "validate duplicate export error output"
```
Comment (dummdidumm) [inline packages/kit/src/packaging/test/errors/duplicate-export/package.json:4]: `private:true`

File: packages/kit/src/packaging/test/errors/no-lib-folder/package.json (hunk 3 above)
Comment (ignatiusmb) [reply packages/kit/src/packaging/test/errors/no-lib-folder/package.json:5]: Oh yeah, added to the remaining fixtures as well

**Comment by jcalfee** [comment]
Thanks .. I followed the help link to [packaging](https://kit.svelte.dev/docs/packaging) and read all of that, but I'm not sure how to disable typescript so I don't need svelte2tsx and typescript.

**Pull request merged**